sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/TooManyParameters.vb

```diff
@@ -0,0 +1,110 @@
+﻿Imports System
+Imports System.Runtime.InteropServices
+
+Public Class TooManyParameters
+  Implements MyInterface
+
+  Public Sub New(ByVal p1 As Integer, ByVal p2 As Integer, ByVal p3 As Integer)
+  End Sub
+
+  Public Sub New(ByVal p1 As Integer, ByVal p2 As Integer, ByVal p3 As Integer, ByVal p4 As Integer) ' Noncompliant {{Constructor has 4 parameters, which is greater than the 3 authorized.}}
+'               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  End Sub
+
+  Public Sub F1(ByVal p1 As Integer, ByVal p2 As Integer, ByVal p3 As Integer) Implements MyInterface.F1
+  End Sub
+
+  Public Sub F2(ByVal p1 As Integer, ByVal p2 As Integer, ByVal p3 As Integer, ByVal p4 As Integer)  Implements MyInterface.F2 ' Compliant, interface implementation
+  End Sub
+
+  Public Sub F1(ByVal p1 As Integer, ByVal p2 As Integer, ByVal p3 As Integer, ByVal p4 As Integer, ByVal p5 As String) ' Noncompliant {{Sub has 5 parameters, which is greater than the 3 authorized.}}
```

Comment:
Please add some tests with `ByRef`

Reply:
I was actually meaning that it would be nice to have a failing test with `ByRef`. See suggestion.

Reply:
I added tests with `ByRef` at lines 87 and 88 (the issues are raised on the interface definitions)

Reply:
I missed them! I cannot find any mixing though.